src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -943,57 +972,75 @@ internal virtual void PrepareSession()
                 WebSession.RetryIntervalInSeconds = RetryIntervalSec;
             }
         }
-        
+
         internal virtual HttpClient GetHttpClient(bool handleRedirect)
         {
-            HttpClientHandler handler = new();
-            handler.CookieContainer = WebSession.Cookies;
-            handler.AutomaticDecompression = DecompressionMethods.All;
+            var handler = WebSession.Handler;
```

Comment:
I think it should be a method. Since WebSession is a store for (all) parameters we can move all (below) logic in WebSession. This assumes handleRedirect is in WebSession too.
```c#
        internal virtual HttpClient GetHttpClient(bool handleRedirect)
            => WebSession.GetHttpClient();
```

Reply:
Have forwarded GetHttpClient() through to WebSession as suggested. Note that to log warnings when the connection changes I needed to pass out a variable indicating the client was recreated.

I have added tests that rely on the LogWarning() message to verify the connection handling is working as expected.